cyfronet-fid/sat4envi
# [#262] Modal Module

## What is in this PR?

Modal Module, without styling (@abacz should get to it in the next week). This Module integrates with AkitaStore, so that modals can be persisted between reloads during development.

This module contains following components:

* ModalOutlet - responsible for creating and removing modals depending on the store's state
* ModalService, ModalQuery, ModalStore - Store integrations
* AlertModal - used as an alert modal service
* ConfirmModal - used as an confirm modal service
* GenericModalComponent - component containing all boilerplate template for modal (header, content, footer)
* DummyModalComponent - component which can be referenced by developers when developing new modals

## How to test?

There should be a new button in the bottom right corner, which will launch series of modals (first confirm, on accepting, alert, and lastly dummy modal)

![Screenshot 2019-11-28 at 17 01 52](https://user-images.githubusercontent.com/13235269/69820839-598f0080-1202-11ea-8230-cfb416fae951.png)

![Screenshot 2019-11-28 at 17 01 58](https://user-images.githubusercontent.com/13235269/69820395-4e87a080-1201-11ea-9ded-8a47345c7ad6.png)

![Screenshot 2019-11-28 at 17 02 03](https://user-images.githubusercontent.com/13235269/69820405-521b2780-1201-11ea-9038-8c959b7c1581.png)

## WIP

There is a couple of tests missing, but otherwise it is ready for review.

Closes #262
